lib/core/app/policies/ros/application_policy.rb

```diff
@@ -43,32 +43,17 @@ def initialize(user, record)
     end
 
     # UserPolicy.new({ policies: ['IamFullAccess'] }, nil).index?
-    def index?
-      check_action(:index?)
-    end
-
-    def show?
-      check_action(:show?)
-    end
-
-    def create?
-      check_action(:create?)
-    end
-
-    def new?
-      create?
-    end
-
-    def update?
-      check_action(:update?)
-    end
-
-    def edit?
-      update?
+    def method_missing(m, *args, &blocks)
```

Comment:
Suggestion instead of `method_missing`:
```
%i[new index show create update edit destroy].define_method do |method|
    check_action(method) # or send("#{method}?")
```

Reply:
Than maybe we should use another `state_machine` library, which validates both transitions and state changes? for example: [this one](https://github.com/state-machines/state_machines-activerecord)
In this case we can keep REST style and use:
``` PUT .../campaign/1  -d {status: 'active'}```

Reply:
@alexeylightit I had a discussion with @rjayroach, and we agreed to proceed with the way you suggested (`define_method` instead of `method_missing`).

Thanks!